v2/internal/attacktechniques/azure/persistence/bastion-shareable-link/main.go

```diff
@@ -0,0 +1,182 @@
+
+` + codeBlock + `json hl_lines="7"
+{
+  {
+    "category": {
+        "value": "Administrative",
+        "localizedValue": "Administrative"
+    },
+    "level": "Informational",
+    "operationName": {
+        "value": "Microsoft.Network/bastionHosts/createshareablelinks/action",
+        "localizedValue": "Creates shareable urls for the VMs under a bastion and returns the urls"
+    },
+    "resourceGroupName": "stratus-red-team-shareable-link-rg-tz6o",
+    "resourceProviderName": {
+        "value": "Microsoft.Network",
+        "localizedValue": "Microsoft.Network"
+    },
+    "resourceType": {
+        "value": "Microsoft.Network/bastionHosts",
+        "localizedValue": "Microsoft.Network/bastionHosts"
+    },
+    "resourceId": "[removed]/resourceGroups/stratus-red-team-shareable-link-rg-tz6o/providers/Microsoft.Network/bastionHosts/stratus-red-team-shareable-link-bastion-tz6o",
+    "status": {
+        "value": "Succeeded",
+        "localizedValue": "Succeeded"
+    },
+    "subStatus": {
+        "value": "",
+        "localizedValue": ""
+    },
+    "properties": {
+        "eventCategory": "Administrative",
+        "entity": "[removed]/resourceGroups/stratus-red-team-shareable-link-rg-tz6o/providers/Microsoft.Network/bastionHosts/stratus-red-team-shareable-link-bastion-tz6o",
+        "message": "Microsoft.Network/bastionHosts/createshareablelinks/action",
+        "hierarchy": "[removed]"
+    },
+}
+` + codeBlock + `
+`,
+		Platform:                   stratus.Azure,
+		IsSlow:                     true,
+		IsIdempotent:               false,
+		MitreAttackTactics:         []mitreattack.Tactic{mitreattack.Persistence},
+		PrerequisitesTerraformCode: tf,
+		Detonate:                   detonate,
+		Revert:                     revert,
+	})
+}
+
+func detonate(params map[string]string, providers stratus.CloudProviders) error {
+	bastionName := params["bastion_name"]
+	resourceGroup := params["resource_group_name"]
+	vmId := params["vm_id"]
+	vmName := params["vm_name"]
+	tenantId := params["tenant_id"]
+
+	ctx := context.Background()
+	cred := providers.Azure().GetCredentials()
+	subscriptionID := providers.Azure().SubscriptionID
+	clientOptions := providers.Azure().ClientOptions
+
+	client, err := armnetwork.NewClientFactory(subscriptionID, cred, clientOptions)
+	if err != nil {
+		log.Fatalf("failed to create client: %v", err)
+	}
+
+	// Create Bastion shareable link
+	// Reference method: https://learn.microsoft.com/en-us/rest/api/virtualnetwork/put-bastion-shareable-link/put-bastion-shareable-link
+	log.Println("Getting Bastion shareable link for VM " +vmName)
+
+	poller, err := client.NewManagementClient().BeginPutBastionShareableLink(ctx, resourceGroup, bastionName, armnetwork.BastionShareableLinkListRequest{
+		VMs: []*armnetwork.BastionShareableLink{
+			{
+				VM: &armnetwork.VM{
+					ID: to.Ptr(vmId),
+				},
+			},
+		},
+	}, nil)
+	if err != nil {
+		log.Fatalf("failed to create shareable link: %v", err)
```

Comment:
we generally try to return errors here (i.e. `return fmt.Errorf("failed to create shareable link: %v", err);`)